examples/api/utils.cpp

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Abdalrhman Mohamed
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Implementations of utility methods.
+ */
+
+#include "utils.h"
+
+#include <iostream>
+
+/**
+ * Get the string version of define-fun command.
+ * @param f the function to print
+ * @param params the function parameters
+ * @param body the function body
+ * @return a string version of define-fun
+ */
+std::string defineFunToString(const cvc5::api::Term& f,
+                              const std::vector<cvc5::api::Term> params,
+                              const cvc5::api::Sort& sort,
+                              const cvc5::api::Term body)
+{
+  std::stringstream ss;
+  ss << "(define-fun " << f << " (";
+  std::vector<cvc5::api::Term>::const_iterator it = params.cbegin(),
+                                               cend = params.cend();
+  if (it != cend)
```

Comment:
These 4 lines are spurious, if you remove them I think it is equivalent.

Reply:
Could you elaborate?

Reply:
This is essentially just hard coding one run of the while loop below, if you delete lines 36-40, the code will be the same I believe.

Reply:
They're almost identical. Line 43 prints 1 more space character than line 38.

Reply:
Replaced with a for loop.